### DISCUSSION TITLE
Import error at setting COMMENTS_XTD_MODEL

### DISCUSSION THREAD

**Discussion by scorpion-HF** [danirus/django-comments-xtd#306] [Q&A]
I have an app named commenting and a model named BookComment
so I set COMMENTS_XTD_MODEL='commenting.models.BookComment'
and got this error: 
ImportError: Module "commenting.models" does not define a "BookComment" attribute/class
and hear my commenting.models.py:
from django_comments_xtd.models import XtdComment
from django.db import models

class BookComment(XtdComment):
    pass
what can I do?
can it be a classic circular dependency problem?

**Comment by danirus** [comment]
Does the `commenting` directory have an `__init__.py`?
Did you add `commenting` to your `INSTALLED_APPS`?
Does it have an `app.py` module? With what content?
Check that, and maybe you could share your app directory. Also the examples directory contains a demo that extends django_comments_xtd in the same way you want to do it, [here](https://github.com/danirus/django-comments-xtd/tree/master/example/custom).

**Reply by scorpion-HF** [reply]
yes commenting directory has __init__.py
I did add commenting to my INSTALLED_APPS
It also has app.py like this:

from django.apps import AppConfig

class CommentingConfig(AppConfig):
    name = 'commenting'
and here is my project directory :
book_store
  |
  book_store
  catalog
  commenting
    |
    migrations
    templates
    __init__.py
    apps.py
    models.py
    views.py
    admin.py
    tests.py
    urls.py
  users
  media
this is my project repository you can check it:
https://github.com/scorpion-HF/book_store.git
I wandted to override get_absolute_url of comment model so I was not successful so I overrode a view of the original app 
you can see it at my project

**Reply by danirus** [reply]
If you see the call stack when it fails you can see that the problem is produced by the imports in the `models.py` module. If you comment the imports that you don't need in `commenting.models` you fix the problem.

Good luck with the project! ;-)
